src/core/actions/creators.ts

```diff
@@ -535,13 +535,13 @@ namespace ActionCreators {
   }
 
   /**
-   * Sets the details product in the store, doing some additional emits and state changes.
+   * Sets the details product in the store with an optional template.
    * @param  {Record | Store.Product}         product - The product to use as the details product.
    * @param  {Template}         template - The template to use as the details template.
-   * @return {Actions.SetDetails}         - Action with details.
+   * @return {Actions.UpdateDetails}         - Action with details.
    */
-  export function setDetails(product: Record | Store.Product, template?: Template): Actions.SetDetails {
-    return createAction(Actions.SET_DETAILS, { data: product, template: SearchAdapter.extractTemplate(template) });
+  export function setDetails(product: Record | Store.Product, template?: Template): Actions.UpdateDetails {
```

Comment:
If `setDetails` just returns an `updateDetails` action we should just deprecate/remove `setDetails`, make sure clients using it fix their code, and fix where we use `setDetails`